common/persistence/pinotVisibilityTripleManager.go

```diff
@@ -290,91 +309,240 @@ func (v *pinotVisibilityTripleManager) chooseVisibilityManagerForWrite(ctx conte
+	}
+	randNum := rand.Intn(10)
+	if randNum != 5 { // Intentionally to have 1/10 chance to log custom query parameters
+		return
+	}
+
+	v.logger.Info("Logging user query parameters for Pinot/ES response comparator...",
+		tag.OperationName(userParam.operation),
+		tag.WorkflowDomainName(userParam.domainName),
+		tag.WorkflowType(userParam.workflowType),
+		tag.WorkflowID(userParam.workflowID),
+		tag.WorkflowCloseStatus(userParam.closeStatus),
+		tag.VisibilityQuery(userParam.customQuery))
+}
+
 func (v *pinotVisibilityTripleManager) ListOpenWorkflowExecutions(
 	ctx context.Context,
 	request *ListWorkflowExecutionsRequest,
 ) (*ListWorkflowExecutionsResponse, error) {
+	v.logUserQueryParameters(userParameters{
+		operation:   string(Operation.LIST),
+		domainName:  request.Domain,
+		closeStatus: -1, // is open. Will have --open flag in comparator workflow
+	}, request.Domain)
+
 	manager := v.chooseVisibilityManagerForRead(request.Domain)
+	if override := ctx.Value("visibility-override"); override == Primary {
+		manager = v.esVisibilityManager
+	} else if override == Secondary {
+		manager = v.pinotVisibilityManager
+	}
 	return manager.ListOpenWorkflowExecutions(ctx, request)
 }
 
 func (v *pinotVisibilityTripleManager) ListClosedWorkflowExecutions(
 	ctx context.Context,
 	request *ListWorkflowExecutionsRequest,
 ) (*ListWorkflowExecutionsResponse, error) {
+	v.logUserQueryParameters(userParameters{
+		operation:   string(Operation.LIST),
+		domainName:  request.Domain,
+		closeStatus: 0, // is closed. Will not have --open flag in comparator workflow
+	}, request.Domain)
 	manager := v.chooseVisibilityManagerForRead(request.Domain)
+	if override := ctx.Value("visibility-override"); override == Primary {
+		manager = v.esVisibilityManager
+	} else if override == Secondary {
+		manager = v.pinotVisibilityManager
+	}
 	return manager.ListClosedWorkflowExecutions(ctx, request)
 }
 
 func (v *pinotVisibilityTripleManager) ListOpenWorkflowExecutionsByType(
 	ctx context.Context,
 	request *ListWorkflowExecutionsByTypeRequest,
 ) (*ListWorkflowExecutionsResponse, error) {
+	v.logUserQueryParameters(userParameters{
+		operation:    string(Operation.LIST),
+		domainName:   request.Domain,
+		workflowType: request.WorkflowTypeName,
+		closeStatus:  -1,
+	}, request.Domain)
 	manager := v.chooseVisibilityManagerForRead(request.Domain)
+	if override := ctx.Value("visibility-override"); override == Primary {
+		manager = v.esVisibilityManager
+	} else if override == Secondary {
+		manager = v.pinotVisibilityManager
+	}
 	return manager.ListOpenWorkflowExecutionsByType(ctx, request)
 }
 
 func (v *pinotVisibilityTripleManager) ListClosedWorkflowExecutionsByType(
 	ctx context.Context,
 	request *ListWorkflowExecutionsByTypeRequest,
 ) (*ListWorkflowExecutionsResponse, error) {
+	v.logUserQueryParameters(userParameters{
+		operation:    string(Operation.LIST),
+		domainName:   request.Domain,
+		workflowType: request.WorkflowTypeName,
+		closeStatus:  0,
```

Comment:
When we list closed workflows, you set the closeStatus to 0. Do you know if it should include the terminated or cancelled workflows whose closeStatus is not 0.

Reply:
What I was thinking is: as we are converting to a CLI cmd in the comparator workflow, we only need the status to be either open or closed (Either need the `--open` flag or not). So, as long as it is one of the closed status, it should be okay. The only API that cares about closeStatus is `ListXXXByStatus`.

Reply:
OK, so you mean when clostStatus != 0 or 1, then we know it is ListXXXByStatus?